style-guide.md

```diff
@@ -0,0 +1,408 @@
+# Pulumi Frontend Style Guide
+
+## Intro
+At Pulumi, we have a lot of frontend surfaces - our marketing website, docs, Service console, and HubSpot, among others.  This style guide is meant as a collection of recommendations for code across all the surfaces.
+
+The goals of this guide are:
+- Create consistency across the many different frontends we have
+- Offer a starting point for folks who make frontend changes less frequently
+- Lower the amount of discussion needed in PRs about code style decisions
+
+This is a resource for teams - you might choose to use this style guide fully, or intentionally deviate in some way.  
```

Comment:
I love this -- thank you for adding it. ❤️ I care quite a bit about giving people the freedom to make their own choices, and using guides like these to serve more as a helpful reference than as law. ⭐